projects/core/src/routing/facade/routing.service.ts

```diff
@@ -17,9 +18,14 @@ export class RoutingService {
   constructor(
     protected store: Store<RouterState>,
     protected winRef: WindowRef,
-    protected semanticPathService: SemanticPathService
+    protected semanticPathService: SemanticPathService,
+    protected routingParamsService: RoutingParamsService
```

Comment:
If we do it in 2.1, we have to make this dependency optional :-(

_+_ Safely do `?.getParams() || <fallback_observable>` in line 26. Fallback could be `NEVER` or `EMPTY` (it's opinionated).
_+_ deprecating the old constructor 
_+_ plan the schematics ticket for 3.0

Reply:
Good point. We can certainly postpone the release of this feature.